### PR TITLE
chore: Add guide on how to run Docker plugins when running the CLI inside Docker

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Based on https://github.com/cloudquery/cloudquery/issues/17547#issuecomment-2042904677

Fixes https://github.com/cloudquery/cloudquery/issues/17547
Fixes https://github.com/cloudquery/cloudquery/issues/17531

Might need better phrasing and name for the title

### CODE REVIEW

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    configs:
+      - spec.yml
+    depends_on:
+      airtable:
+        condition: service_healthy
+      postgres:
+        condition: service_healthy
+  airtable:
+    container_name: airtable
+    image: docker.cloudquery.io/cloudquery/source-airtable:v2.1.2
```
Comment (bbernays) [inline website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:60]: Is there a standardized way that users can know/find  the docker image so this can be relevant to all dockerized plugins?

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+# Running Docker Plugins With the CLI Docker Image
```
Comment (bbernays) [inline website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:1]:
```suggestion
# Running Docker Plugins
```

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md (hunk 1 above)
Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:60]: The path is `docker.cloudquery.io/<team>/<kind>-<name>:<version>`

Each plugin has this documented

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  tables: ["*"]
+  destinations: ["postgresql"]
+---
+kind: destination
+spec:
+  name: "postgresql"
+  path: "cloudquery/postgresql"
+  version: "VERSION_DESTINATION_POSTGRESQL"
+```
+3. Run `cloudquery plugin install spec.yml` to pull the Docker plugin image locally.
```
Comment (bbernays) [inline website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:32]: Could we just have them run `docker pull <image>`?

Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:32]: No that won't work as our registry is private. Also `docker login docker.cloudquery.io` won't work as our registry requires you to pass a team name in a special header https://github.com/cloudquery/plugin-pb-go/blob/78b50f093a96d903467634c085f3be46dcfaf8db/managedplugin/docker.go#L121
There's a way to configure the additional headers but it's a bit complicated

**Review by pilvikala — COMMENTED** [review]
I think I expected something a bit different (not using plugins as containers in a docker-compose) but to my surprise, we did not have this one either so it's actually quite a good improvement to our docs!

I think this is struggling a bit between trying to make this very basic (like opening postgres ports while specifying easy password in plain text) and too advanced (with healthchecks and stuff). I'd recommend to choose one approach and keep it. I'd start with the basic thing to provide the minimum example with a callout that this is an example to show how it's supposed to work and should not be used as is in production.
In short, I'd cut the dependencies, restrictions on pulling

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    depends_on:
+      airtable:
+        condition: service_healthy
+      postgres:
+        condition: service_healthy
+  airtable:
+    container_name: airtable
+    image: docker.cloudquery.io/cloudquery/source-airtable:v2.1.2
+    # We use `cloudquery login` and `cloudquery plugin install spec.yml` to pull the image locally
+    pull_policy: never
```
[the remark this answers is not in the document]
Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:62]: Explained in https://github.com/cloudquery/cloudquery/pull/17581#discussion_r1559613735, only our CLI can pull from our private registry since we pass the team name

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:32]: Only our CLI can pull from our private registry since we pass the team name

**Comment by erezrokah** [comment]
> I think I expected something a bit different (not using plugins as containers in a docker-compose) but to my surprise, we did not have this one either so it's actually quite a good improvement to our docs!
> 
> I think this is struggling a bit between trying to make this very basic (like opening postgres ports while specifying easy password in plain text) and too advanced (with healthchecks and stuff). I'd recommend to choose one approach and keep it. I'd start with the basic thing to provide the minimum example with a callout that this is an example to show how it's supposed to work and should not be used as is in production. In short, I'd cut the dependencies, restrictions on pulling, just to focus on the core here.

This is the minimal example :) Without the health checks the CLI would start syncing before the plugin or the DB is ready and fail. The pull restrictions are required since we need to pass a team name to our registry as a part of the authentication flow (we need to know which team downloads which plugin).

As for the approach - to run multiple containers you can use compose or k8s which is more complicated I think. Maybe there's a simpler approach I'm not aware of (other than "docker in docker").

Please note this is doc is under the "advanced" section of our docs

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md (hunk 3 above)
Comment (bbernays) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:32]: It doesn't seem that hard to specify the custom header... https://docs.docker.com/engine/reference/commandline/cli/#custom-http-headers

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
+# Running Docker Plugins
```
Comment (erezrokah) [inline website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:1]:
```suggestion
# Using CloudQuery Docker plugins inside a containerized environment 
```

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md (hunk 3 above)
Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:32]: You need to modify your `config.json` and the headers are sent will all requests to all registries I believe

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md]: Changed it to `Using CloudQuery Docker Registry Plugins Inside a Containerized Environment`

File: website/pages/docs/advanced-topics/docker-cli-docker-plugins.md (hunk 3 above)
Comment (erezrokah) [reply website/pages/docs/advanced-topics/docker-cli-docker-plugins.md:32]: We also pass the plugin version since docker doesn't https://github.com/cloudquery/plugin-pb-go/blob/78b50f093a96d903467634c085f3be46dcfaf8db/managedplugin/docker.go#L120 which is dynamic and we can't put it in the `config.json`

**Review by pilvikala — APPROVED** [review]
